app/components/progress_bar_component.rb

```diff
@@ -0,0 +1,26 @@
+class ProgressBarComponent < ViewComponent::Base
+    PROG_TITLE_TO_GROUPINGS = {
+      "Teach primary computing" => [
+        "Contribute to an online discussion",
+        "Develop your teaching practice",
+        "Develop computing in your community"
+      ],
+      "Teach secondary computing" => [
+        "Develop your subject knowledge",
+        "Develop your teaching practice",
+        "Develop computing in your community"
+      ]
+    }.freeze
+  
+    def initialize(current_user, programme)
+      @programme = programme
+      @class_names = {}
```

Comment:
```suggestion
```

Reply:
I think this is still needed

Reply:
The refactorings look good - thank you.

Can you explain here why you chose not to refactor the class names code? It would really help me reviewing to know your reasons (even if you discussed them with Barry) - might also help future maintainers of this code.

Reply:
Sorry yes you are correct, factored thank you